client/src/pages/index.js

```diff
@@ -11,7 +13,16 @@ const defaultState = {
 function Index() {
   const [state, setState] = useState(defaultState)
   const activeTheme = useSyncLocalStorage('activeTheme')
+  const auth = useSyncV('auth')
+  const router = useRouter()
 
+  useEffect(()=>{
+    if (auth.authUser) {
+      router.push('/contacts')
+    } else {
+      router.push('/login')
+    }
+  })
```

Comment:
Looks good, but I found a few inconsistencies for the case:

> when visiting /
> - signed in user will be redirected to contacts/

Please view the attached video and take note of the multiple redirects on the URL (URL route goes from "/" to "/login" and finally, "/contacts"), and the red-letter warnings on the chrome dev console.

### Video: Multiple Redirection and Chrome Dev Warnings

https://user-images.githubusercontent.com/56998001/235442888-9adff0da-13e5-4e9b-903b-d8dd6df362ed.mp4

---

### Suggestions

Here are some suggestions but please confirm and verify if it's effective:

Listen for the `auth.authLoading` status in the deps array and call `router.push` only after `auth.authLoading` becomes `false`.

```
  useEffect(()=>{
    if (!auth.authLoading) {
      if (auth.authUser) {
        router.push('/contacts')
      } else {
        router.push('/login')
      }
    }
  }, [auth])
```